hack/alpha-build-machinery/make/targets/openshift/controller-gen.mk

```diff
@@ -1,29 +1,22 @@
 self_dir :=$(dir $(lastword $(MAKEFILE_LIST)))
 
-CONTROLLER_GEN_VERSION ?=v0.2.1
-CONTROLLER_GEN_TEMP ?=$(PERMANENT_TMP_GOPATH)/src/sigs.k8s.io/controller-tools
-controller_gen_gopath =$(shell realpath -m $(CONTROLLER_GEN_TEMP)/../..)
-CONTROLLER_GEN ?=$(CONTROLLER_GEN_TEMP)/controller-gen
+CONTROLLER_GEN ?=$(PERMANENT_TMP_GOPATH)/bin/controller-gen
```

Comment:
why not give it a unique name? This way, when switching branches, you easily end up with the wrong generator.

Reply:
`controller-gen-$(CONTROLLER_GEN_VERSION)`

Reply:
Honestly, when switching branches `make clean` is needed, and `make update` in this repo is not that lengthy so I don't think that makes it worth the effort.

Reply:
Remember Tim Hockin's twitter comment about make semantics? Having to run make clean is such a violation.